src/cargo/ops/cargo_rustc/custom_build.rs

```diff
@@ -406,6 +406,17 @@ impl BuildOutput {
         }
         Ok((library_paths, library_links))
     }
+
+    /// Sort the contents of the struct for consistent hashing. 
+    /// Suggested if populated from a HashMap instead of an order-preserving data source
+    pub fn sort(&mut self){
+        self.library_paths.sort();
+        self.library_links.sort();
```

Comment:
This and `library_paths` above I believe can be quite significant in terms of ordering, would it be possible to preserve the original orderings rather than sorting these?